doc/source/notebooks/intro_to_gpflow2.pct.py

```diff
@@ -174,42 +175,92 @@ def noisy_sin(x):
 model
 
 # %% [markdown]
-# ## Training using Gradient Tapes
+# ## Training using training_loss and training_loss_closure
+#
+# GPflow models come with training_loss and training_loss_closure methods to make it easy to train your models.
+# There is a slight difference between models that own their own data (most of them, e.g. GPR, VGP, ...) and models that do not own the data (SVGP).
 #
-# In TensorFlow 2, we can optimize (trainable) model parameters with TensorFlow optimizers using `tf.GradientTape`. In this simple example, we perform one gradient update of the Adam optimizer to minimize the negative marginal log likelihood (or ELBO) of our model.
+# ### Model-internal data
+# For models that own their own data (inheriting from InternalDataTrainingLossMixin), data is provided at model construction time.
+# In this case, model.training_loss does not take any arguments, and can be directly passed to an optimizer's `minimize()` method:
+
+# %%
+gpr_model = gpflow.models.VGP(data, kernel, likelihood)
+optimizer = tf.optimizers.Adam()
+optimizer.minimize(
+    gpr_model.training_loss, gpr_model.trainable_variables
+)  # Note: this does a single step
+# In practice, you will need to call minimize() many times, this will be further discussed below.
+
+# %% [markdown]
+# This also works for the Scipy optimizer, though it will do the full optimization on a single call to minimize():
+
+# %%
+optimizer = gpflow.optimizers.Scipy()
+optimizer.minimize(
+    gpr_model.training_loss, gpr_model.trainable_variables, options=dict(maxiter=ci_niter(1000))
+)
+
+# %% [markdown]
+# You can obtain a compiled version using training_loss_closure, whose `jit` argument is True by default:
+
+# %%
+gpr_model.training_loss_closure()  # compiled
+gpr_model.training_loss_closure(jit=True)  # compiled
+gpr_model.training_loss_closure(jit=False)  # uncompiled, same as gpr_model.training_loss
+
+# %% [markdown]
+# The SVGP model inherits from ExternalDataTrainingLossMixin and expects the data to be passed to training_loss():
+
+# %%
+assert isinstance(model, gpflow.models.SVGP)
+model.training_loss(data)
+
+# %% [markdown]
+# To make optimizing it easy, it has a `training_loss_closure()` method, that takes the data and returns a closure that computes the training loss on this data:
```

Comment:
would be nice to add a sentence saying what `data` is in this example. A `tf.dataset` or `np.ndarray` tuple?

Reply:
tried adding a sentence, let me know what you think